packages/core/src/types.ts

```diff
@@ -735,7 +811,11 @@ export interface StateMachine<
   TContext,
   TStateSchema extends StateSchema,
   TEvent extends EventObject,
-  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
+  TTypestate extends Typestate<TContext> = { value: any; context: TContext },
+  _TAction extends ActionObject<TContext, TEvent> = ActionObject<
```

Comment:
Is there any value in adding this here now if it isn't used anywhere?

Reply:
Not at the moment; this will become useful when we want to strongly type actions in `State` (separate PR)